### PR TITLE
Add potential fix for cleaning up builds

### PR DESCRIPTION
**Pull request by Myoldmopar**
This is essentially the change proposed by @rraustad for cleaning up Windows builds.  They have deployed with this fix and their Windows builds now come up as expected.

### CODE REVIEW

File: ci.rb [hunk 1]
```diff
@@ -335,12 +335,15 @@ def get_limits(t_options, t_client, t_repo)
                   regression_baselines << [compiler, regression_base];
 
                   if !File.directory?(regression_base.get_build_dir(compiler))
```
Comment (rraustad) [inline ci.rb]: This (!File.directory) means that the folder is NOT there. So removing (rm) it on line 339 will not change anything.

You have to remove the folder before you get here.

                  if File.directory?(regression_base.get_build_dir(compiler))
                    $logger.info "Removing pre-existing regression build directory (#{regression_base.get_build_dir(compiler)})"
                    system("rm -rf #{regression_base.get_build_dir(compiler)}")
                  end

Comment (rraustad) [reply ci.rb]: So I suggest moving the new lines down into the else and bracketing with if(keep_build_folder) print "skipping" else remove folder.

Comment (rraustad) [reply ci.rb]: ...else remove folder and do lines 340-342.

Comment (Myoldmopar) [reply ci.rb]: This PR became a **crazy** overhaul of Decent CI.  Much slimmer now, and a new build folder setup which will enable ccache capability to speed up builds.  Folders should be cleaned up nicely between every compiler run now.  You can check out the new process here: https://github.com/Myoldmopar/decent_ci/blob/d08606fdb7434b8fc56b75c5cb4fc468d62bbeb4/ci.rb#L274-L316

**Comment by Myoldmopar** [comment]
OK.  I'm going to pull the trigger and merge this in.  We can always revert but I don't think we'll need to.  If we find issues, we'll probably just push forward and deal with it there rather than going back to the past.  I think all the CI machines are currently shut down, so I'll bring them up in a controlled manner.  :crossed_fingers:

**Pull request merged**